src/server/system_services/bucket_server.js

```diff
@@ -1107,22 +1109,20 @@ function get_cloud_buckets(req) {
                 });
                 const used_cloud_buckets = cloud_utils.get_used_cloud_targets(['AWS', 'S3_COMPATIBLE', 'FLASHBLADE', 'IBM_COS'],
                     system_store.data.buckets, system_store.data.pools, system_store.data.namespace_resources);
-                return P.ninvoke(s3, "listBuckets")
-                    .timeout(EXTERNAL_BUCKET_LIST_TO)
+                return P.timeout(EXTERNAL_BUCKET_LIST_TO, P.ninvoke(s3, 'listBuckets'))
```

Comment:
```suggestion
                return P.timeout(EXTERNAL_BUCKET_LIST_TO, s3.listBuckets().promise())
```

Reply:
Hey, notice that I intentionally didn't go over all these calls and left deprecated implementations until we can complete the async-await change.
```
// deprecated
exports.fromCallback = fromCallback; // 96 occurences
exports.fcall = fcall; // 64 occurences
exports.ninvoke = ninvoke; // 32 occurences
exports.wait_until = wait_until; // 21 occurences
exports.Defer = Defer; // 18 occurences
exports.pwhile = pwhile; // 17 occurences
```